src/core/drive/page_renderer.ts

```diff
@@ -16,9 +16,7 @@ export class PageRenderer extends Renderer<HTMLBodyElement, PageSnapshot> {
 
   finishRendering() {
     super.finishRendering()
-    if (this.isPreview) {
-      this.focusFirstAutofocusableElement()
-    }
+    this.focusFirstAutofocusableElement()
```

Comment:
@javan @sstephenson what are your thoughts on this change? What's the incentive for not autofocusing elements when navigating to a page that is not a preview?

Reply:
I think it's backwards, currently. It used to be https://github.com/hotwired/turbo/blob/269fa2436495a720ceb67215dfa8abe4d4143492/src/core/drive/page_renderer.ts#L45-L47 and got flipped around (accidentally?) in https://github.com/hotwired/turbo/commit/a7d02f0fde3d95d6e3af9ec72b7654d3fbeedd33.

Reply:
@javan oooh, great digging, thank you.

Just so that I'm clear in my understanding when I revert this change, do we want:

* autofocus on navigation
* do not autofocus when visiting a reviewable page
  * i.e. autofocus element on `/a`, navigate to `/b`, navigate back to `/a` without autofocusing)
* do not autofocus when navigating through History.goBack()

Reply:
I can't manage to concoct a test to cover skipping autofocus during previews, so I'll share my attempts here for posterity's sake:

```diff
diff --git a/src/core/drive/page_renderer.ts b/src/core/drive/page_renderer.ts
index 6a83d45..94d2a3e 100644
--- a/src/core/drive/page_renderer.ts
+++ b/src/core/drive/page_renderer.ts
@@ -16,7 +16,9 @@ export class PageRenderer extends Renderer<HTMLBodyElement, PageSnapshot> {
 
   finishRendering() {
     super.finishRendering()
-    this.focusFirstAutofocusableElement()
+    if (!this.isPreview) {
+      this.focusFirstAutofocusableElement()
+    }
   }
 
   get currentHeadSnapshot() {
diff --git a/src/core/drive/visit.ts b/src/core/drive/visit.ts
index 7ba354a..1187f55 100644
--- a/src/core/drive/visit.ts
+++ b/src/core/drive/visit.ts
@@ -234,7 +234,7 @@ export class Visit implements FetchRequestDelegate {
       const isPreview = this.shouldIssueRequest()
       this.render(async () => {
         this.cacheSnapshot()
-        await this.view.renderPage(snapshot)
+        await this.view.renderPage(snapshot, isPreview)
         this.adapter.visitRendered(this)
         if (!isPreview) {
           this.complete()
diff --git a/src/tests/fixtures/autofocus.html b/src/tests/fixtures/autofocus.html
index dfb1354..a4f2a35 100644
--- a/src/tests/fixtures/autofocus.html
+++ b/src/tests/fixtures/autofocus.html
@@ -21,5 +21,7 @@
     <turbo-frame id="drives-frame" target="frame">
       <a id="drives-frame-target-link" href="/src/tests/fixtures/frames/form.html">#drives-frame link to frames/form.html</a>
     </turbo-frame>
+
+    <a id="navigation-link" href="/src/tests/fixtures/navigation.html">navigation.html</a>
   </body>
 </html>
diff --git a/src/tests/functional/autofocus_tests.ts b/src/tests/functional/autofocus_tests.ts
index 5db02f0..c44004a 100644
--- a/src/tests/functional/autofocus_tests.ts
+++ b/src/tests/functional/autofocus_tests.ts
@@ -19,6 +19,19 @@ export class AutofocusTests extends TurboDriveTestCase {
     this.assert.notOk(await this.hasSelector("#second-autofocus-element:focus"), "focuses the first [autofocus] element on the page")
   }
 
+  async "test does not autofocus on preview visit"() {
+    await this.clickSelector("#navigation-link")
+    await this.nextBody
+    await this.clickSelector("#autofocus-link")
+
+    this.assert.notOk(await this.hasSelector("#first-autofocus-element:focus"), "does not autofocus when previewed")
+    this.assert.notOk(await this.hasSelector("#second-autofocus-element:focus"), "does not autofocus when previewed")
+
+    await this.nextBody
+
+    this.assert.ok(await this.hasSelector("#first-autofocus-element:focus"), "autofocuses after discarding the preview")
+  }
+
   async "test navigating a frame with a descendant link autofocuses [autofocus]:first-of-type"() {
     await this.clickSelector("#frame-inner-link")
     await this.nextBeat

```